File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        int n=nums.size();
        int first=n-1;
        for( int i=n-2 ; i>=0 ; i-- ){
            if( nums[i]+i >= first ){
                first=i;
            }
        }
        if( first == 0 ){
            return true;
        }
        return false;
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        
        int n=nums.length, flag=n-1;
        for( int i=n-2; i>=0 ; i-- ){
            if( nums[i]+i >= flag ){
                flag = i;
            }
        }
        if( flag == 0 ){
            return true;
        }
        return false;
    }
}